README.md

```diff
@@ -230,248 +75,244 @@ const pseudorandomBits = hkdf.generateBits({
+``` mermaid
+%%{init: {'flowchart': {'curve': 'linear'}}}%%
 
-NB this is the first step of the double ratchet / KDF, as explained in [§1.5.1.1](#double-ratchet-initialization).
+flowchart LR
+    XCC["HKDF(X25519 Shared Secret) = (IV, Key)"]
+    ReqTempSK["Requestor Temp\nX25519 Secret Key"]
 
-```
-                 Payload
-
-          ┌────────KDF────────┐
-          │                   │
-          │  256-bit AES & IV │
-          │         │         │
-          └─────────┼─────────┘
-                    │
-                    ▼
-┌────────────────AES-GCM───────────────┐
-│                                      │
-│  ┌──────────────UCAN──────────────┐  │
-│  │                                │  │
-│  │  iss: Responder                │  │
-│  │  aud: ReqECDH                  │  │
-│  │  att: []                       │  │
-│  │  fct: [nextResECDH, challenge] │  │
-│  │  prf: ...                      │  │
-│  │                                │  │
-│  └────────────────────────────────┘  │
-│                                      │
-└──────────────────────────────────────┘
+    subgraph Payload
+        ProvTempPK["Provider's Temp\nX25519 DID"]
+
+        subgraph Env [Encrypted Envelope]
+            ucan["UCAN
+                iss: ProviderDID&nbsp;&nbsp;&nbsp;&nbsp;
+                aud: ReqECDH&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+                att: []&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+                fct: [challenge]&nbsp;&nbsp;&nbsp;
+                prf: ...&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+            "]
+        end
+    end
+
+    ReqTempSK -..->|1/2| XCC
+    ProvTempPK -.->|1/2| XCC
+    XCC --->|decrypts| Env
 ```
 
 Upon receipt, the Requestor MUST validate that the UCAN capabilities in the proof fulfill their `caps` criteria. The UCAN itself MUST be valid, unrevoked, unexpired, and intended for the temporary DID (the `aud` field). If any of these checks fail, the session MUST be abandoned, the temporary DID regenerated, and the protocol restarted from [intention broadcast](#32-requestor-broadcasts-intent).
 
-### 3.3.1 Validation UCAN
+### 5.2.1 Payload
 
-The validation UCAN MUST NOT be used to delegate any capabilities. This UCAN MUST only be used to prove access to capabilities and sign the next ECDH public key. The `att` and `my` fields MUST be empty arrays. The issuer (`iss`) field MUST contain the Responder's long-term DID (rather than the temporary ECDH DID). The audience (`aud`) field MUST contain the Requestor's temporary ECDH DID from [§3.2](#32-requestor-broadcasts-intent).
+To start the Double Ratchet, the payload in this stage has the highest number of cleartext fields. Note that the value in the `iss` field contain the temporary ECDH DIDs, and MUST NOT use the Provider's actual long-term DID. Conversely, the UCAN inside the encrypted payload MUST use the Provider's long-term DID.
```

Comment:
Hmm yeah. I think it could even be improved to make it more RFC 2119:

```suggestion
To start the Double Ratchet, the payload in this stage has the highest number of cleartext fields. Note that the value in the `iss` field MUST contain the temporary ECDH DIDs, and MUST NOT use the Provider's actual long-term DID. Conversely, the UCAN inside the encrypted payload MUST use the Provider's long-term DID.
```